### PR TITLE
Add support of truffle test --migrate-none and event message #3887

### PR DESCRIPTION
**Pull request by dongmingh**

### CODE REVIEW

File: packages/core/lib/testing/Test.js [hunk 1]
```diff
@@ -244,7 +244,7 @@ const Test = {
     const migrateConfig = config.with({
       reset: true,
       resolver: resolver,
-      quiet: true
+      quiet: false
```
Comment (gnidan) [inline packages/core/lib/testing/Test.js:247]: Should we set this to `false` if-and-only-if `--migrate-none` is turned on? `truffle test` silences migrate output on purpose right now, and I don't think we want to change that most of the time.

I do like your use of this mechanism, though! Let the migrate system do its own logging!

File: packages/migrate/Migration.js [hunk 2]
```diff
@@ -185,6 +185,14 @@ class Migration {
    * @param  {Object}   options  config and command-line
    */
   async run(options) {
+    // check --migrate-none
+    if (config.migrateNone || config['migrate-none']) {
+        if (config.events) {
+            config.events.emit("migrate:nothingToMigrate");
```
Comment (gnidan) [inline packages/migrate/Migration.js:191]: I know you copied this from `truffle test --compile-none`, but I think we should use a better message here, e.g. `migrate:skipped`, which would say something like "> Migrations have been skipped, not migrating."

(Maybe there should be an issue to fix this for the compile case)

**Review by gnidan — COMMENTED** [review]
Here's a few small comments; would be good for others to review also!

File: packages/core/lib/testing/Test.js (hunk 1 above)
Comment (haltman-at) [reply packages/core/lib/testing/Test.js:247]: Yeah I'm also confused by this; I don't think we should be setting it to `false` at all?  Presumably we want to suppress the printout regardless.

**Comment by haltman-at** [comment]
A note on the git usage here -- you seem to have forked the repo and are making a PR from that fork?  That's a bit troublesome for other people to get at.  (Like, I wanted to check out your branch to inspect it, but I couldn't, because it's on another repo.)  The thing to do instead is to simply make a branch, on the main repo, and PR that branch, not create a whole fork of the repo.  Feel free to ask about this if you're not sure how that works.

File: packages/migrate/Migration.js [hunk 3]
```diff
@@ -185,6 +185,14 @@ class Migration {
    * @param  {Object}   options  config and command-line
    */
   async run(options) {
+    // check --migrate-none
+    if (config.migrateNone || config['migrate-none']) {
```
Comment (haltman-at) [inline packages/migrate/Migration.js:189]: Style note: We use 2-space indenting here, not 4. :)  We used to have `prettier` handle this sort of thing for you, but, uh, husky hasn't been working in quite a while (#3530), so we're stuck doing it ourselves for now. :-/  I believe @cds-amal was working on fixing this, but I'm not sure what the current status of that is.

File: packages/core/lib/testing/Test.js (hunk 1 above)
Comment (dongmingh) [reply packages/core/lib/testing/Test.js:247]: will change it to false if and only if --migrate-none

File: packages/events/defaultSubscribers/migrate.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const OS = require("os");
+
+module.exports = {
+  initialization: function () {
+    this.logger = console;
+  },
+  handlers: {
+    "migrate:start": [
```
Comment (haltman-at) [inline packages/events/defaultSubscribers/migrate.js:8]: OK, something really does not make sense here.  You've added a handler for `"migrate:start"`... but you haven't added anything to actually *emit* this event, nor have you removed any existing code that would perform this printout manually.  What's going on here?

File: packages/migrate/Migration.js (hunk 2 above)
Comment (dongmingh) [reply packages/migrate/Migration.js:191]: will change accordingly

File: packages/migrate/Migration.js [hunk 5]
```diff
@@ -185,6 +185,14 @@ class Migration {
...
+        if (config.events) {
```
Comment (haltman-at) [inline packages/migrate/Migration.js:190]: Is this check necessary?  I don't think we check this in other places, do we?  We can assume that `config.events` exists, right?

**Review by haltman-at — COMMENTED** [review]
Reviewing this a bit early, normally I'd wait until it's passing CI. :P

**Comment by haltman-at** [comment]
Btw, if your PR is meant to address a specific issue (#3887), you should mention that issue in the text, not just the title, because if you only put it in the title, it doesn't become a clickable link.

...however, you want to avoid using keywords that will trigger GitHub's auto-close thing, because we don't use that here. :)  (We close issues when the fix is released, rather than when it's merged.)  So, uh, be careful of that... I like to say things like "Addresses #3887" because that won't trip GitHub's system. :)

File: packages/events/defaultSubscribers/migrate.js (hunk 4 above)
Comment (dongmingh) [reply packages/events/defaultSubscribers/migrate.js]: remove migrate:start

File: packages/migrate/Migration.js (hunk 5 above)
Comment (dongmingh) [reply packages/migrate/Migration.js:190]: yes, config.events exists, remove the if-statement
